### PR TITLE
MM-24677: Stage group configuration changes.

### PR DESCRIPTION
**Pull request by mkraft**
#### Summary

Stage all group configuration changes and commit them upon save.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-24677

### CODE REVIEW

File: sass/components/_groups.scss [hunk 1]
```diff
@@ -725,7 +725,8 @@
 }
 
 .icon-over-input {
-    padding-left: 4px;
-    padding-top: 8px;
+    padding-left: 12px;
+    padding-top: 9px;
     position: absolute;
+    opacity: .8;
```
Comment (mkraft) [inline sass/components/_groups.scss:731]: These are some unrelated fixes to the group at-mention name change styling.

File: components/admin_console/group_settings/group_details/group_details.jsx [hunk 2]
```diff
@@ -86,6 +92,16 @@ export default class GroupDetails extends React.PureComponent {
...
 
+    componentDidUpdate(prevProps) {
+        /* eslint-disable react/no-did-update-set-state */
+        if (this.props.groupTeams !== prevProps.groupTeams) {
+            this.setState({groupTeams: this.props.groupTeams});
+        }
+        if (this.props.groupChannels !== prevProps.groupChannels) {
+            this.setState({groupChannels: this.props.groupChannels});
+        }
+    }
```
Comment (mkraft) [inline components/admin_console/group_settings/group_details/group_details.jsx:103]: According to [the React docs](https://reactjs.org/docs/react-component.html#componentdidupdate) this is fine, not sure why eslint is configured to not allow it.

File: e2e/cypress/integration/system_console/group_configuration_spec.js [hunk 3]
```diff
@@ -0,0 +1,435 @@
...
+                savePage();
+
+                // * Check the groupteam via the API to ensure its role wasn't updated
+                cy.apiGetGroupChannel(groupID, channel.id).then(({body}) => {
+                    expect(body.scheme_admin).to.eq(false);
+                });
+            });
+        });
+    });
+});
```
Comment (mkraft) [inline e2e/cypress/integration/system_console/group_configuration_spec.js:435]: These take 7.5 minutes to run!

File: e2e/cypress/integration/system_console/group_configuration_spec.js [hunk 4]
```diff
@@ -0,0 +1,435 @@
...
+                cy.apiLinkGroupChannel(groupID, channel.id);
+
+                // # Reload the page
+                cy.visit(`/admin_console/user_management/groups/${groupID}`);
+
+                // * Check that the channel was added to the view
+                teamOrChannelIsPresent(channel.display_name);
+
+                // # Click remove
+                cy.get(`button[data-testid='${channel.display_name}_groupsyncable_remove']`).click();
```
Comment (fmunshi) [inline e2e/cypress/integration/system_console/group_configuration_spec.js:202]:
```suggestion
                cy.findByTestId(`${channel.display_name}_groupsyncable_remove']`).click();
```

Comment (fmunshi) [reply e2e/cypress/integration/system_console/group_configuration_spec.js:202]: Theres a few other places where you could this could be changed too, not a big deal though

File: e2e/cypress/support/api_commands.js [hunk 5]
```diff
@@ -1281,6 +1300,95 @@ Cypress.Commands.add('apiPatchGroup', (groupID, patch) => {
...
+        url: `/api/v4/groups/${groupID}/${syncableType}s/${syncableID}`,
+        method: 'GET',
+        timeout: 60000,
+    }).then((response) => {
+        expect(response.status).to.equal(200);
+        return cy.wrap(response);
+    });
+}
+
+function getGroupSyncables(groupID, syncableType) {
```
Comment (fmunshi) [inline e2e/cypress/support/api_commands.js:1351]: These commands need comments describing the parameters

**Comment by amyblais** [comment]
@mkraft FYI this hasn't been cherry-picked yet.

**Comment by mkraft** [comment]
> @mkraft FYI this hasn't been cherry-picked yet.

Done now. Thanks Amy.
